lib/dma.c

```diff
@@ -623,16 +644,39 @@ dma_controller_dirty_page_get(dma_controller_t *dma, vfu_dma_addr_t addr,
          * around.
          */
         if (val == 0) {
-            *outp = 0;
+            out = 0;
         } else {
             uint8_t zero = 0;
             __atomic_exchange(&region->dirty_bitmap[i], &zero,
-                              outp, __ATOMIC_SEQ_CST);
+                              &out, __ATOMIC_SEQ_CST);
```

Comment:
99.9% of the time we'll be the exact same page size: consider only doing this when we have to
(otherwise make outp direct to bitmap as before)